test/jit/test_tracer.py

```diff
@@ -2367,3 +2367,41 @@ def forward(
         with self.assertRaisesRegex(RuntimeError, "cannot be understood by the tracer, only outputs matching"):
             mod = ReturnsBadDict()
             traced_module = torch.jit.trace(mod, [torch.ones(1), torch.ones(1)], strict=False)
+
+    def test_trace_func_argument_names_captured(self):
+        def fn(first_arg: torch.Tensor, second_arg: torch.Tensor) -> torch.Tensor:
+            return first_arg + second_arg
+
+        traced_fn = torch.jit.trace(fn, (torch.ones(1), torch.ones(1)))
+        FileCheck().check("first_arg").check_next("second_arg") \
+            .run(str(traced_fn.graph))
+
+    def test_trace_partial_func_argument_names_captured(self):
```

Comment:
Consider adding a case in which none of them are captured. In other words, is this part being tested:

```
        if not param.kind == param.POSITIONAL_OR_KEYWORD:
            return []
```

Reply:
Done. Added in a separate test file as it doesn't really belong here.